ShinerNFT.sol

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.7;
+
+import "erc721a/contracts/ERC721A.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+
+contract OwnableDelegateProxy {}
+contract ProxyRegistry { mapping(address => OwnableDelegateProxy) public proxies; }
+
+contract LocalTest is ERC721A, Ownable
+{
+	uint256 public constant MAX_SUPPLY = 11111;
+	uint256 public constant MINT_PRICE = 0.069 ether;
+	address internal constant _community = 0x36E92eF696ac7f7B7Cf7BA9EC0C7D42E6e65A8AD;
+	string internal _baseTokenURI = 'https://fake.robocosmic.online/';
+	address immutable proxyRegistryAddress;
+	bytes32 public _merkleRoot;
+
+	enum Status { Initial, Presale, Sale, SoldOut }
+	Status public status = Status.Initial;
+
+	event StatusChanged();
+
+	constructor(
+		bytes32 merkleRoot,
+		address _proxyRegistryAddress
+	) ERC721A("LocalTest", "LTS")
+	{
+		_merkleRoot = merkleRoot;
+		proxyRegistryAddress = _proxyRegistryAddress;
+
+		_mint(_community, 13, '', false); // Initial Mint
+	}
+
+	modifier onlyAccounts() { require(msg.sender == tx.origin, "NO_ALLOWED_ORIGIN"); _; }
+
+	function mint(uint256 quantity)
+		external
+		payable
+		onlyAccounts
+	{
+		require(status == Status.Sale, "SALE_CLOSED");
+		require(quantity != 0, "INVALID_QUANTITY");
+		require(quantity <= 8, "QUANTITY_EXCEEDS_MAX_PER_CALL");
+		require(quantity * MINT_PRICE == msg.value, "WRONG_ETH_AMOUNT");
+		require(totalSupply() + quantity <= MAX_SUPPLY, "MAX_SUPPLY_EXCEEDED");
```

Comment:
Use `_totalMinted()` instead of `totalSupply()`.



While the logic is correct   
(`_totalMinted()` is always equal to `totalSupply()` when the token cannot be burned)

`_totalMinted()` can save a little gas because it does not subtract the total burn count internally.

Reply:
Only the `_burn` and `burn` functions can transfer the NFT to the zero address.

If you check against `_totalMinted()`, then it won’t allow for extra minting after tokens have been burned.